File: CS/dxSample_IAnyControl/Grid Data/DataHelper.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;

namespace dxSample_IAnyControl.IAny_Files {
    public class DataHelper {
        public static object GetData() {
            DataTable dt = new DataTable();
            dt.Columns.Add("PersonID", typeof(int));
            dt.Columns.Add("PersonFirstName", typeof(string));
            dt.Columns.Add("PersonLastName", typeof(string));
            dt.Columns.Add("PersonTitle", typeof(string));
            dt.Columns.Add("HireDate", typeof(DateTime));
            dt.Columns.Add("HomePhone", typeof(string));
            dt.Columns.Add("SalesByProductCategory", typeof(object));
            FillDataTable(dt);
            return dt;
        }
        internal static void FillDataTable(DataTable dt) {
            dt.Rows.Add(new object[] { 0, "Nancy", "Davolio", "Sales Representative", "5/1/2005", "(206) 555-9857", GetEmployeeSellingsByCategoryChartDataEx(10) });
            dt.Rows.Add(new object[] { 1, "Andrew", "Fuller", "Sales Representative", "5/1/2005", "(206) 555-9857", GetEmployeeSellingsByCategoryChartDataEx(7) });
            dt.Rows.Add(new object[] { 2, "Mike", "Livingston", "Sales Representative", "5/1/2005", "(206) 555-9857", GetEmployeeSellingsByCategoryChartDataEx(5) });
            dt.Rows.Add(new object[] { 3, "Margared", "Peacock", "Sales Representative", "5/1/2005", "(206) 555-9857", GetEmployeeSellingsByCategoryChartDataEx(1) });
        }
        private static List<EmployeeSellingByCategoryChartData> GetEmployeeSellingsByCategoryChartDataEx(int key) {
            List<EmployeeSellingByCategoryChartData> res = new List<EmployeeSellingByCategoryChartData>();
            res.Add(new EmployeeSellingByCategoryChartData() {
                CategoryName = "Beverages", Quantity = key * 200
            });
            res.Add(new EmployeeSellingByCategoryChartData() {
                CategoryName = "Condiments", Quantity = key * 50
            });
            res.Add(new EmployeeSellingByCategoryChartData() {
                CategoryName = "Confections", Quantity = key * 100
            });
            res.Add(new EmployeeSellingByCategoryChartData() {
                CategoryName = "Diary products", Quantity = key * 90
            });
            res.Add(new EmployeeSellingByCategoryChartData() {
                CategoryName = "Grans / Cereals", Quantity = key * 150
            });
            res.Add(new EmployeeSellingByCategoryChartData() {
                CategoryName = "Meat / Poultry", Quantity = key * 30
            });
            res.Add(new EmployeeSellingByCategoryChartData() {
                CategoryName = "Produce ", Quantity = key * 120
            });
            res.Add(new EmployeeSellingByCategoryChartData() {
                CategoryName = "Seafood ", Quantity = key * 90
            });
            return res;
        }
    }
}

File: VB/dxSample_IAnyControl/Grid Data/DataHelper.vb
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Linq

Namespace dxSample_IAnyControl.IAny_Files
	Public Class DataHelper
		Public Shared Function GetData() As Object
			Dim dt As New DataTable()
			dt.Columns.Add("PersonID", GetType(Integer))
			dt.Columns.Add("PersonFirstName", GetType(String))
			dt.Columns.Add("PersonLastName", GetType(String))
			dt.Columns.Add("PersonTitle", GetType(String))
			dt.Columns.Add("HireDate", GetType(Date))
			dt.Columns.Add("HomePhone", GetType(String))
			dt.Columns.Add("SalesByProductCategory", GetType(Object))
			FillDataTable(dt)
			Return dt
		End Function
		Friend Shared Sub FillDataTable(ByVal dt As DataTable)
			dt.Rows.Add(New Object() { 0, "Nancy", "Davolio", "Sales Representative", "5/1/2005", "(206) 555-9857", GetEmployeeSellingsByCategoryChartDataEx(10) })
			dt.Rows.Add(New Object() { 1, "Andrew", "Fuller", "Sales Representative", "5/1/2005", "(206) 555-9857", GetEmployeeSellingsByCategoryChartDataEx(7) })
			dt.Rows.Add(New Object() { 2, "Mike", "Livingston", "Sales Representative", "5/1/2005", "(206) 555-9857", GetEmployeeSellingsByCategoryChartDataEx(5) })
			dt.Rows.Add(New Object() { 3, "Margared", "Peacock", "Sales Representative", "5/1/2005", "(206) 555-9857", GetEmployeeSellingsByCategoryChartDataEx(1) })
		End Sub
		Private Shared Function GetEmployeeSellingsByCategoryChartDataEx(ByVal key As Integer) As List(Of EmployeeSellingByCategoryChartData)
			Dim res As New List(Of EmployeeSellingByCategoryChartData)()
			res.Add(New EmployeeSellingByCategoryChartData() With {.CategoryName = "Beverages", .Quantity = key * 200})
			res.Add(New EmployeeSellingByCategoryChartData() With {.CategoryName = "Condiments", .Quantity = key * 50})
			res.Add(New EmployeeSellingByCategoryChartData() With {.CategoryName = "Confections", .Quantity = key * 100})
			res.Add(New EmployeeSellingByCategoryChartData() With {.CategoryName = "Diary products", .Quantity = key * 90})
			res.Add(New EmployeeSellingByCategoryChartData() With {.CategoryName = "Grans / Cereals", .Quantity = key * 150})
			res.Add(New EmployeeSellingByCategoryChartData() With {.CategoryName = "Meat / Poultry", .Quantity = key * 30})
			res.Add(New EmployeeSellingByCategoryChartData() With {.CategoryName = "Produce ", .Quantity = key * 120})
			res.Add(New EmployeeSellingByCategoryChartData() With {.CategoryName = "Seafood ", .Quantity = key * 90})
			Return res
		End Function
	End Class
End Namespace
